src/shared/sam/cli/samCliInfo.ts

```diff
@@ -18,13 +14,16 @@ export class SamCliInfoInvocation {
     public constructor(private readonly samPath: string) {}
 
     public async execute(): Promise<SamCliInfoResponse> {
-        const childProcessResult = await new ChildProcess(this.samPath, ['--info'], { logging: 'no' }).run()
+        const r = await new ChildProcess(this.samPath, ['--info'], { logging: 'no' }).run()
 
-        logAndThrowIfUnexpectedExitCode(childProcessResult, 0)
-        const response = this.convertOutput(childProcessResult.stdout)
+        if (r.exitCode !== 0) {
+            // getVersionValidatorResult() will return `SamCliVersionValidation.VersionNotParseable`.
+            return { version: '' }
```

Comment:
No need to throw here.

Reply:
Isn't it better to fail early though? If we aren't going to throw then I think it makes more sense to return `undefined` instead of an empty value because it forces callers to deal with the difference.

Reply:
yes, in this case the caller is `getVersionValidatorResult()` and it signals "undefined" as `VersionNotParseable`